File: frontend-web/publisher-app/src/app/shared/models/Post.ts
export class Post {
    id: number;
    title: string;
    content: string;
    author: string;
    postStatus : string;
    creationDate: string;

    constructor(title: string, content: string, author: string, status: string, id: number, creationDate: string) {
        this.title = title;
        this.content = content;
        this.author = author;
        this.postStatus = status;
        this.id = id;
        this.creationDate = creationDate;
    }
}

File: backend-java/SpringCloudPXL/post-service/src/main/java/be/pxl/services/domain/Post.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String title;
    @Column(columnDefinition = "LONGTEXT")
    private String content;
    private String author;
    private LocalDate creationDate;
    @Enumerated(EnumType.STRING)
    private PostStatus status;

}
